### ISSUE TITLE
NullPointerException - Collections.binarySearch while querying for edges

### ISSUE THREAD

**Issue by pankajydv** [JanusGraph/janusgraph#357]
`(new MultiVertexCentricQueryBuilder((StandardTitanTx) trans)).(vertices).labels(GraphConstants.M_PUBLISH).direction(Direction.IN).edges();
`

The above query fails with NullPointerException sometimes (it behaves correctly at times) in case if there are no incident edges for the input.

```
java.lang.NullPointerException
        at java.util.Collections.binarySearch(Collections.java:212)
        at com.thinkaurelius.titan.diskstorage.keycolumnvalue.SliceQuery.getSubset(SliceQuery.java:94)
        at com.thinkaurelius.titan.graphdb.vertices.CacheVertex.loadRelations(CacheVertex.java:55)
        at com.thinkaurelius.titan.graphdb.query.vertex.SimpleVertexQueryProcessor.getBasicIterator(SimpleVertexQueryProcessor.java:111)
        at com.thinkaurelius.titan.graphdb.query.vertex.SimpleVertexQueryProcessor.iterator(SimpleVertexQueryProcessor.java:68)
```

The exception mentioned above is for titan-cassandra-1.0.0, but is reproducible for janusgraph as well.

**Comment by amcp** [comment]
@pankajydv can you create repro in a PR in the janusgraph-test module please?

**Comment by pankajydv** [comment]
I have not been able to reproduce it consistently, however putting this NULL check in `CacheVertex.loadRelations(final SliceQuery query, final Retriever<SliceQuery, EntryList> lookup)` has resolved this issue for me. I get this NullPointer exception for following query, when there is no edge for the following query

`(new MultiVertexCentricQueryBuilder((StandardTitanTx) trans)).(vertices).labels(GraphConstants.M_PUBLISH).direction(Direction.IN).edges();
`


```
            if (superset == null || superset.getValue() == null) {
                result = lookup.get(query);
            } else {
                result = query.getSubset(superset.getKey(), superset.getValue());
            }
```

**Comment by pankajydv** [comment]
Created a pull request, #362. Request to please review and merge

**Comment by twilmes** [comment]
@pankajydv have you been able to reproduce this issue without using multi-query, ie. the `BasicVertexCentricQueryBuilder`? I'm hesitant to close this issue out with your PR because it doesn't get to root cause of the NPE. When you have reproduced this locally, were you just running that query you provided over and over? I know it was intermittent, but can you describe a little more about how you had seen it locally previously?

**Comment by porunov** [comment]
Fixed by #1581
